daphne/src/roles_test.rs

```diff
@@ -4,39 +4,117 @@
 use crate::{
     auth::BearerToken,
     constants::{MEDIA_TYPE_AGG_INIT_REQ, MEDIA_TYPE_AGG_SHARE_REQ, MEDIA_TYPE_COLLECT_REQ},
-    hpke::{HpkeDecrypter, HpkeReceiverConfig},
+    hpke::HpkeReceiverConfig,
     messages::{
         AggregateInitializeReq, AggregateResp, AggregateShareReq, CollectReq, HpkeCiphertext, Id,
         Interval, Nonce, Report, ReportShare, TransitionFailure, TransitionVar,
     },
     roles::{DapAggregator, DapHelper, DapLeader},
-    testing::{MockAggregator, HPKE_RECEIVER_CONFIG_LIST, LEADER_BEARER_TOKEN},
-    DapAbort, DapError, DapMeasurement, DapRequest, Prio3Config, VdafConfig,
+    testing::{
+        BucketInfo, MockAggregator, ReportStore, HPKE_RECEIVER_CONFIG_LIST, LEADER_BEARER_TOKEN,
+    },
+    DapAbort, DapMeasurement, DapRequest, Prio3Config, VdafConfig,
 };
 use assert_matches::assert_matches;
 use prio::codec::{Decode, Encode};
-use rand::prelude::*;
-use std::vec;
+use std::{ops::DerefMut, vec};
 
-#[tokio::test]
-async fn http_post_aggregate_unauthorized_request() {
-    let mut rng = thread_rng();
-    let helper = MockAggregator::new();
-    let task_id = helper.nominal_task_id();
-    let task_config = helper.get_task_config_for(task_id).unwrap();
+fn gen_agg_init_req(
+    agg: &MockAggregator,
+    report_shares: Vec<ReportShare>,
+    sender_auth: Option<BearerToken>,
```

Comment:
I think it would be cleaner to drop this input parameter and force the caller to add sender auth as needed.

Reply:
Sounds good. Callers of these functions are now responsible for adding `sender_auth` now.